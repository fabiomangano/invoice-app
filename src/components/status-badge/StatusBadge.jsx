import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './StatusBadge.style.scss';

export const STATUS = {
  PENDING: 'pending',
  PAID: 'paid',
  DRAFT: 'draft',
};

const StatusBadge = ({ children, variant }) => (
  <div className={cn('status-badge', `status-badge__background--${variant}`)}>
    <div
      className={cn(`status-badge__circle`, `status-badge__circle--${variant}`)}
    />
    <span
      className={(`status-badge__label`, `status-badge__label--${variant}`)}
    >
      {children}
    </span>
  </div>
);

StatusBadge.propTypes = {
  children: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['paid', 'pending', 'draft']).isRequired,
};

export default StatusBadge;
