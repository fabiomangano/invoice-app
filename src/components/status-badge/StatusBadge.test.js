import React from 'react';
import Renderer from 'react-test-renderer';
import StatusBadge from './StatusBadge';

describe('StatusBagde', () => {
  let tree;

  afterEach(() => {
    tree = null;
  });

  it('should render a badge for the paid status', () => {
    Renderer.act(() => {
      tree = Renderer.create(<StatusBadge variant="paid">Paid</StatusBadge>);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render a badge for the pending status', () => {
    Renderer.act(() => {
      tree = Renderer.create(
        <StatusBadge variant="pending">Pending</StatusBadge>,
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render a badge for the draft status', () => {
    Renderer.act(() => {
      tree = Renderer.create(<StatusBadge variant="draft">Draft</StatusBadge>);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
