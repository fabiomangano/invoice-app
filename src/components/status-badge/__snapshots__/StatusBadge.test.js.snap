// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBagde should render a badge for the draft status 1`] = `
<div
  className="status-badge status-badge__background--draft"
>
  <div
    className="status-badge__circle status-badge__circle--draft"
  />
  <span
    className="status-badge__label--draft"
  >
    Draft
  </span>
</div>
`;

exports[`StatusBagde should render a badge for the paid status 1`] = `
<div
  className="status-badge status-badge__background--paid"
>
  <div
    className="status-badge__circle status-badge__circle--paid"
  />
  <span
    className="status-badge__label--paid"
  >
    Paid
  </span>
</div>
`;

exports[`StatusBagde should render a badge for the pending status 1`] = `
<div
  className="status-badge status-badge__background--pending"
>
  <div
    className="status-badge__circle status-badge__circle--pending"
  />
  <span
    className="status-badge__label--pending"
  >
    Pending
  </span>
</div>
`;
